testing/test_junitxml.py

```diff
@@ -1,20 +1,52 @@
 import os
 import platform
 from datetime import datetime
+from pathlib import Path
 from xml.dom import minidom
 
 import py
+import xmlschema
 
 import pytest
 from _pytest.junitxml import LogXML
 from _pytest.reports import BaseReport
 
 
-def runandparse(testdir, *args):
-    resultpath = testdir.tmpdir.join("junit.xml")
-    result = testdir.runpytest("--junitxml=%s" % resultpath, *args)
-    xmldoc = minidom.parse(str(resultpath))
-    return result, DomNode(xmldoc)
+@pytest.fixture(scope="session")
+def schema():
+    """Returns a xmlschema.XMLSchema object for the junit-10.xsd file"""
+    fn = Path(__file__).parent / "example_scripts/junit-10.xsd"
+    with fn.open() as f:
+        return xmlschema.XMLSchema(f)
+
+
+@pytest.fixture(name="run_and_parse")
+def run_and_parse_(testdir, schema):
```

Comment:
Why the indirection via `name`?

Reply:
I was replacing `run_and_parse` call by a fixture, this ensures I didn't make any mistakes during development (this is the intention for the `name` parameter to `fixture`).